src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+import mate.jdbc.util.ConnectionUtil;
+
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (title, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement
+                        = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            insertStatement.setString(1, car.getTitle());
+            insertStatement.setLong(2, car.getManufacturer().getId());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setId(id);
+            }
+            addDriversToCar(car);
+            return car;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create a car: " + car, throwable);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id, c.title, c.manufacturer_id, m.title, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ? AND c.is_deleted = false";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getCarStatement
+                                = connection.prepareStatement(query)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car.setId(id);
+                car.setManufacturer(getManufacturerBySet(resultSet));
+                car.setTitle(resultSet.getString(2));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a car by id " + id, throwable);
+        }
+        car.setDrivers(getDriversById(id));
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id, c.title, c.manufacturer_id, m.title, m.country "
+                + "FROM cars c JOIN manufacturer m ON c.manufacturer_id = m.id"
+                + " WHERE c.is_deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement getDriverStatement
+                    = connection.prepareStatement(query);
+            ResultSet resultSet = getDriverStatement.executeQuery();
+            Car car = null;
+            getDriverStatement.close();
+            connection.close();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setDrivers(getDriversById(id));
```

Comment:
don't open a connection inside another opened connection
use `for-each` after `catch` block and set drivers for each car in your `cars` list